### PR TITLE
Silenceable Automatics

### PR DESCRIPTION
**Pull request by FulminatingGold**
Requested by El Presidente.

## About The Pull Request
All automatics can have a silencer attached.

## Why It's Good For The Game
Automatics should be silenceable.

## Changelog
:cl:
tweak: automatics accept silencers
/:cl:

### CODE REVIEW

**Comment by zombiemimic** [comment]
the silenceable guns have sprites for it, so unless you're willing to sprite them all, I don't think this will go through

**Comment by SirRichardFrancis** [comment]
image
Copypasting silencer is easy enough, but there isn't any space for it on a sprite, is there?

**Comment by thedarkcrusader** [comment]
Wait they do i never noticed  any sprites on the stray light

**Comment by SirRichardFrancis** [comment]
> Wait they do i never noticed any sprites on the stray light

image
There is silenced sprite for straylight. It's not visible in game or what?

**Comment by Sparrowbird611** [comment]
Straylight has it, I'm unsure if any other guns do. Not sure if it really matters if it has a sprite tbh, most other gun attachments don't.

**Comment by zombiemimic** [comment]
> Straylight has it, I'm unsure if any other guns do. Not sure if it really matters if it has a sprite tbh, most other gun attachments don't.

drozd, makarov, atreides, clarissa, and giskard all have silenced sprites, I'm sure there are more

**Comment by thedarkcrusader** [comment]
> > Wait they do i never noticed any sprites on the stray light
> 
> image
> There is silenced sprite for straylight. It's not visible in game or what?

There's no sprire of it in game

**Comment by thedarkcrusader** [comment]
> Straylight has it, I'm unsure if any other guns do. Not sure if it really matters if it has a sprite tbh, most other gun attachments don't.

Im preying to god it does not matter

File: code/modules/projectiles/guns/projectile/automatic/vintorez.dm [hunk 1]
```diff
@@ -26,6 +26,7 @@
 		FULL_AUTO_400
 		)
 
+	gun_tags = list(GUN_SILENCABLE)
```
Comment (Firefox13) [inline code/modules/projectiles/guns/projectile/automatic/vintorez.dm:29]: It already comes silenced

File: code/modules/projectiles/guns/projectile/automatic/lmg.dm [hunk 2]
```diff
@@ -27,6 +27,8 @@
 	rarity_value = 80
 	gun_parts = list(/obj/item/part/gun = 1 ,/obj/item/stack/material/plasteel = 4)
 
+	gun_tags = list(GUN_SILENCABLE)
```
Comment (Firefox13) [inline code/modules/projectiles/guns/projectile/automatic/lmg.dm:30]: LMGs should have no business being silenced. Remove it

File: code/modules/projectiles/guns/projectile/automatic/motherfucker.dm [hunk 3]
```diff
@@ -28,6 +28,8 @@
 	var/recentpumpmsg = 0
 	var/pumped = FALSE
 
+	gun_tags = list(GUN_SILENCABLE)
```
Comment (Firefox13) [inline code/modules/projectiles/guns/projectile/automatic/motherfucker.dm:31]: It's six-barrelled. *No.* Unless you can think of a way to make it consume six of those.

File: code/modules/projectiles/guns/projectile/automatic/z8.dm [hunk 4]
```diff
@@ -31,6 +31,8 @@
 		list(mode_name="fire grenades", mode_desc="Unlocks the underbarrel grenade launcher", burst=null, fire_delay=null, move_delay=null,  icon="grenade", use_launcher=1)
 		)
 
+	gun_tags = list(GUN_SILENCABLE)
```
Comment (Firefox13) [inline code/modules/projectiles/guns/projectile/automatic/z8.dm:34]: I'm pretty sure it will mess with the grenade launcher

Comment (thedarkcrusader) [reply code/modules/projectiles/guns/projectile/automatic/z8.dm:34]: Silenced grenade launcher

File: code/modules/projectiles/guns/projectile/automatic/maxim.dm [hunk 5]
```diff
@@ -34,6 +34,8 @@
 	spawn_blacklisted = TRUE
 	slowdown_hold = 5
 
+	gun_tags = list(GUN_SILENCABLE)
```
Comment (Firefox13) [inline code/modules/projectiles/guns/projectile/automatic/maxim.dm:37]: Also LMG

Comment (thedarkcrusader) [reply code/modules/projectiles/guns/projectile/automatic/maxim.dm:37]: Will do

**Comment by Jan-Blasiak** [comment]
LMGs and other automatics could work if silencers would have durability, and depending on weapon type it would wear of slower or faster.

https://www.youtube.com/watch?v=kbq-FWw8O0w

**Comment by FulminatingGold** [comment]
I disabled silencer for some guns, now suck my fat nuts.

**Pull request merged**